Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -920,3 +926,25 @@ lemma mulEquivOfOrderOfEq_symm_apply_gen : (mulEquivOfOrderOfEq hg hg' h).symm g
 end mulEquiv
 
 end generator
+
+section prod
+
+/-- The product of two finite groups is cyclic iff
+both of them are cyclic and their orders are coprime. -/
+@[to_additive "The product of two finite additive groups is cyclic iff
+both of them are cyclic and their orders are coprime."]
+theorem Group.isCyclic_prod_iff
+    {M N : Type*} [CommGroup M] [Finite M] [CommGroup N] [Finite N] :
+    IsCyclic (M × N) ↔
+      IsCyclic M ∧ IsCyclic N ∧ (Nat.card M).Coprime (Nat.card N) := by
```

Comment:
CommGroup can be generalized to Group but let's leave that to another PR ...

Reply:
Commutativity and finiteness are removed in [this branch](https://github.com/AntoineChambert-Loir/mathlib4/compare/ZModUnits...alreadydone:mathlib4:ZModUnits), which also contain other golfs. The only remaining task is to golf `Group.isCyclic_of_coprime_card_ker` to golf `isCyclic_units_of_prime_pow`.

Reply:
I opened https://github.com/AntoineChambert-Loir/mathlib4/pull/1; I'm not golfing `isCyclic_units_of_prime_pow` because it won't make the proof much shorter.

Reply:
I managed to merge your commit!